src/main/java/hudson/plugins/deploy/DeployPublisher.java

```diff
@@ -102,7 +104,8 @@ public void perform(@Nonnull Run<?, ?> run, @Nonnull FilePath workspace, @Nonnul
 
             FilePath[] wars = workspace.list(warFiles);
             if (wars == null || wars.length == 0) {
-                throw new InterruptedException("[DeployPublisher][WARN] No wars found. Deploy aborted. %n");
+                listener.getLogger().printf("[DeployPublisher][WARN] No wars found. Deploy aborted. %n");
```

Comment:
Unrelated change. AFAICT It changes the behavior of the reporter. Before the change the publisher would have been failing the build, and now it ignores the error. Not sure what should be the right behavior

Reply:
This change was required to get the pipeline syntax tests to run again (the commit which changed the "no WAR found" behavior without updating the tests: https://github.com/jenkinsci/deploy-plugin/commit/3f630c3256fd883f9dc8fa085c6225191725b03c).

So it's either reverting to warning only (what I did) or updating the tests to assert for build failures. I don't know which behavior is better.